Various coding guideline fixes

nashif opened the pull request:
﻿- kernel: Make both operands of operators of same essential type category
- kernel/arch: cleanup function definitions
- userspace: rename _is_user_context -> k_is_user_context
- x86: make tests of a value against zero should be made explicit
- kernel: make tests of a value against zero should be made explicit

nashif committed:
userspace: rename _is_user_context -> k_is_user_context

This functions is being called across the tree, no reason why it should
not be a public API.

The current usage violates a few MISRA rules.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
kernel: Make both operands of operators of same essential type category

Add a 'U' suffix to values when computing and comparing against
unsigned variables and other related fixes of the same MISRA rule (10.4)

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
kernel/arch: cleanup function definitions

make identifiers used in the declaration and definition identical. This
is based on MISRA rule 8.3.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
x86: make tests of a value against zero should be made explicit

Tests of a value against zero should be made explicit, unless the
operand is effectively Boolean. This is based on MISRA rule 14.4.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
kernel: make tests of a value against zero should be made explicit

Tests of a value against zero should be made explicit, unless the
operand is effectively Boolean. This is based on MISRA rule 14.4.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

dcpleung commented on arch/x86/core/x86_mmu.c:
```diff
@@ -1115,10 +1115,10 @@ static pentry_t flags_to_entry(uint32_t flags)
-/* map new region virt..virt+size to phys with provided arch-neutral flags */
-void arch_mem_map(void *virt, uintptr_t phys, size_t size, uint32_t flags)
+/* map new region dest..dest+size to addr with provided arch-neutral flags */
+void arch_mem_map(void *dest, uintptr_t addr, size_t size, uint32_t flags)
 {
-	range_map_unlocked(virt, phys, size, flags_to_entry(flags),
+	range_map_unlocked(dest, addr, size, flags_to_entry(flags),
 			   MASK_ALL, 0);
```
This one is actually better with `phys` and `virt`. Could you change the declaration instead?

dcpleung commented on include/sys/mem_manage.h:
```diff
@@ -270,15 +270,15 @@ void *k_mem_map(size_t size, uint32_t flags);
  */
-size_t k_mem_region_align(uintptr_t *aligned_addr, size_t *aligned_size,
-			  uintptr_t addr, size_t size, size_t align);
+size_t k_mem_region_align(uintptr_t *aligned_phys, size_t *aligned_size,
+			  uintptr_t phys_addr, size_t size, size_t align);
```
`addr` and `aligned_addr` are better as it will align any addresses.

jhedberg commented on kernel/device.c:
```diff
@@ -112,7 +112,7 @@ const struct device *z_impl_device_get_binding(const char *name)
 	 */
-	if ((name == NULL) || (*name == 0)) {
+	if ((name == NULL) || (*name == 0U)) {
 		return NULL;
```
Wouldn’t '\0' be the most appropriate here since it’s a character and not an integer?

jhedberg replied:
It would also be more consistent with the place you’re changing in fatal.c

jhedberg replied:
Seems like you missed this comment? I think the signedness of `char` is also architecture dependent, which is why `\0` would also be better.

nashif committed:
drivers: remove what appears to be code from comments

Some tools consider this as commented code because of the ifdef, so
remove that and fix a coding guideline violation.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
